src/Compilers/CSharp/Portable/BoundTree/BoundDagEvaluation.cs

```diff
@@ -39,8 +39,8 @@ public virtual bool IsEquivalentTo(BoundDagEvaluation other)
                     BoundDagTypeEvaluation e => e.Type,
                     BoundDagDeconstructEvaluation e => e.DeconstructMethod,
                     BoundDagIndexEvaluation e => e.Property,
-                    BoundDagSliceEvaluation e => (Symbol?)e.SliceMethod ?? e.IndexerAccess?.Indexer,
-                    BoundDagIndexerEvaluation e => e.IndexerSymbol ?? e.IndexerAccess?.Indexer,
+                    BoundDagSliceEvaluation e => Binder.GetIndexerOrImplicitIndexerSymbol(e.IndexerAccess),
```

Comment:
>Binder.GetIndexerOrImplicitIndexerSymbol(e.IndexerAccess) [](http://example.com/codeflow?start=49&length=57)

It looks like these two are going to return null for a BoundArrayAccess. It doesn't look like that is going to help IsEquivalentTo implementation and similar usages.